File: pastebook-db/pastebook-db/Models/Album.cs
namespace pastebook_db.Models
{
    public class Album
    {
        public int Id { get; set; }
        public string AlbumName { get; set; } = null!;
        public string AlbumDescription { get; set; } = null!;
        public bool IsPublic { get; set; }
        public bool IsEdited { get; set; }
        public DateTime CreatedOn { get; set; }
        public byte[]? CoverAlbumImage { get; set; }

        // Foreign Key
        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public ICollection<AlbumImage>? AlbumImageList { get; set; }
    }

    public class AlbumDTO
    {
        public int? Id { get; set; }
        public string AlbumName { get; set; } = null!;
        public string AlbumDescription { get; set; } = null!;
        public bool IsPublic { get; set; }
        public bool IsEdited { get; set; }
        public string? CreatedOn { get; set; }
        public byte[]? CoverAlbumImage { get; set; }

        public int UserId { get; set; }

        public ICollection<AlbumImage>? AlbumImageList { get; set; }
    }
}

File: postbook-angular/src/app/models/album.ts
import { User } from "./user";

export class Album {
    id?: string;
    albumName?: string;
    albumDescription?: string;
    isPublic?: boolean;
    isEdited?: boolean;
    createdOn?: string;
    coverAlbumImage?: string;
    userId?: string;

    imageCount?: number;
    imageList?: AlbumImage[];
}

export class AlbumImage {
    id?: string;
    image?: string;
    createdOn?: string;
    isEdited?: boolean;
    likeCount?: number;
    commentCount?: number;
    albumId?: number;
    hasLiked?: boolean;

    albumImageLikesList?: AlbumImageLike[];
    albumImageCommentsList?: AlbumImageComment[];
  }

  export class AlbumImageLike {
    id?: string;

    albumImageId?: string;
    userId?: string;
    user?: User;
  }

  export class AlbumImageComment {
    id?: string;
    comment?: string;
    createdOn?: string;

    albumImageId?: string;
    userId?: string;
    user?: User;
  }